Dockerfile

```diff
@@ -29,4 +31,13 @@ RUN apt-get -yqq update && apt-get -yqq install \
 
 ENV FWROOT=/FrameworkBenchmarks PYTHONPATH=/FrameworkBenchmarks
 
+# Check if Group is already created
+RUN if ! getent group $GROUP_ID; then \
+      addgroup --gid $GROUP_ID user; \
+    fi
+
+# Drop permissions of user to match those of the host system
+RUN adduser --disabled-password --gecos '' --uid $USER_ID --gid $GROUP_ID user
```

Comment:
IMHO you should check if the user exists already too.

Reply:
As long as you're depending on the `ubuntu` images you'll never find a user called `user`.

Reply:
Or do you mean check if the UID exists?

Reply:
It is not the user name that would be the problem, but the user ID.

Reply:
how about I check it like this?

```bash
RUN if ! getent passwd $USER_ID; then \
      adduser --disabled-password --gecos '' --uid $USER_ID --gid $GROUP_ID user; \
    fi
```

Reply:
Hmm actually the next USER step will fail if the user hasn't been created

Reply:
Looks good to me.

Reply:
okay I think I have a fix for that edge-case, see what you think @volyrique

Reply:
That is better, but I think that it can actually be simplified - apparently the `USER` instruction [accepts](https://docs.docker.com/engine/reference/builder/#user) UIDs (and GIDs) as well. So, keep the check, but remove the `else` case, and change the subsequent instruction to:
```
USER $USER_ID
```
If I am not mistaken, that should cover all cases - we don't actually care about the actual username.

Reply:
Updated PTAL